aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -113,21 +110,21 @@ NestedTensorImpl::NestedTensorImpl(
     at::Tensor buffer,
     at::Tensor nested_size_tensor,
     at::Tensor nested_stride_tensor,
-    const std::vector<int64_t>& offsets)
+    std::vector<int64_t> offsets)
     : TensorImpl(
+          Storage(buffer.storage()),
           generate_nested_key_set(buffer),
-          buffer.dtype(),
-          buffer.device()),
-      buffer_(std::move(buffer)),
+          buffer.dtype()),
```

Comment:
Another note is in the wrap_buffer could instead pass in a Storage and a Size, We can then get the device, and dtype from the Storage instance. I am open to this change

Reply:
I think it's a little cleaner to have the `NestedTensorImpl` take a storage directly. `wrap_buffer()` could still take a buffer as a Tensor and pull out the parts to pass to the `NestedTensorImpl` constructor. cc @albanD / @bdhirsh opinion here?

> I am open to this change but would add some more bloat since constructor call sites would need to be changed

Are there many call sites? It seems usually helper functions like `wrap_buffer()` are used instead of the constructor directly

Reply:
Looks like there are 11 calls to: `at::detail::make_tensor<NestedTensorImpl>` 
I think alot of these though are by the transformer folk and should probably switch to `wrap_buffer`.

Reply:
I think this is a good first step. We can always migrate to a unpacked API later if the current one is confusing people.